tests/interop.test.ts

```diff
@@ -1,296 +1,340 @@
+                                    description: 'API stage name of the associated API stage in a usage plan.',
+                                },
+                                throttle: {
+                                    type: 'object',
+                                    additionalProperties: {
+                                        $ref: '#/types/aws-native:apigateway:UsagePlanThrottleSettings',
+                                    },
+                                    description:
+                                        'Map containing method level throttling information for API stage in a usage plan.',
+                                },
+                            },
+                        },
+                        'aws-native:apigateway:UsagePlanThrottleSettings': {
+                            type: 'object',
+                            properties: {
+                                burstLimit: {
+                                    type: 'integer',
+                                },
+                                rateLimit: {
+                                    type: 'number',
+                                },
+                            },
+                        },
+                    },
+                    resources: {
+                        'aws-native:apigateway:UsagePlan': {
+                            cf: 'AWS::ApiGateway::UsagePlan',
+                            inputs: {
+                                apiStages: {
+                                    type: 'array',
+                                    items: {
+                                        $ref: '#/types/aws-native:apigateway:UsagePlanApiStage',
+                                    },
+                                    description: 'The associated API stages of a usage plan.',
+                                },
+                            },
+                        },
+                    },
+                };
             },
-        };
-        const types: { [key: string]: PulumiType } = {};
-        const pulumiProvider = PulumiProvider.AWS_NATIVE;
-
+            { virtual: true },
+        );
         // WHEN
-        const isJson = isJsonType(propName, properties, types, pulumiProvider);
+        const normalized = normalize(
+            {
+                ApiStages: [
+                    {
+                        ApiId: 'api',
+                        Stage: 'stage',
+                        Throttle: {
+                            // These keys should not be normalized, but their properties should
+                            '/v1/toys/GET': {
+                                BurstLimit: 2,
+                                RateLimit: 10,
+                            },
+                        },
+                    },
+                ],
+            },
+            'AWS::ApiGateway::UsagePlan',
+        );
 
         // THEN
-        expect(isJson).toEqual(true);
+        expect(normalized).toEqual({
```

Comment:
Love it. I think the `normalize` tests is a very intuitive place to test, these test read super well.